src/fork_gc.c

```diff
@@ -310,13 +314,38 @@ static void FGC_childCollectTerms(ForkGC *gc, RedisSearchCtx *sctx) {
   FGC_sendTerminator(gc);
 }
 
-static void countDeletedCardinality(const RSIndexResult *r, void *arg) {
-  CardinalityValue *valuesDeleted = arg;
-  for (int i = 0; i < array_len(valuesDeleted); ++i) {
-    if (valuesDeleted[i].value == r->num.value) {
-      valuesDeleted[i].appearances++;
-      return;
+KHASH_MAP_INIT_INT64(cardvals, size_t)
+
+typedef struct {
+  const IndexBlock *lastblk;
+  khash_t(cardvals) * delLast;
+  khash_t(cardvals) * delRest;
```

Comment:
Why not using dict? I do not like this khash, its messy and full of bugs.

Reply:
khash is quicker and simpler in this case. I don't see any real bugs here: https://github.com/attractivechaos/klib/issues

Reply:
I suppose it's possible in theory to use dict in the future, but they didn't have a pre-made hash table for 64 bit integers